subprojects/core/src/main/java/org/gradle/initialization/DefaultGradleLauncher.java

```diff
@@ -201,6 +147,75 @@ public void stop() {
         }
     }
 
+    private class RunBuildAction implements Transformer<BuildResult, BuildOperationContext> {
+        private final Stage upTo;
+
+        private RunBuildAction(Stage upTo) {
+            this.upTo = upTo;
+        }
+
+        @Override
+        public BuildResult transform(BuildOperationContext buildOperationContext) {
+            BuildOperationWorkerRegistry.Completion workerLease = buildServices.get(BuildOperationWorkerRegistry.class).operationStart();
```

Comment:
Using `BuildOperationProcessor` here mean moving the whole processing to another thread. By using `BuildOperationWorkerRegistry` we keep it on the 'Daemon worker' thread. We may not want to change that.